packages/core/src/common/progress-service.ts

```diff
@@ -74,10 +74,7 @@ export class ProgressService {
     async withProgress<T>(text: string, locationId: string, task: () => Promise<T>): Promise<T> {
         const progress = await this.showProgress({ text, options: { cancelable: true, location: locationId } });
         try {
-            const result = task();
```

Comment:
@AlexTugarev `await` was missing, so it will immediately cancel progress

Reply:
thanks!